### PR TITLE
Refactor file-download code to allow AWS etc

### PR DESCRIPTION
**Pull request by drbyte**
- cleaned up the My Account list of downloads, for simpler styling
- changed both admin and catalog to allow observers to provide details of file availability
- added support for AWS-hosted downloads (files on AWS S3) ... simply give `aws:bucketname/filename.ext` as the filename in attributes controller, with expiring links to prevent theft
- added support for HTTP-hosted downloads, such as Dropbox or any other http/https URL which requires no authentication by the customer (offers no theft prevention)
- moved download-by-redirect and download-by-streaming to observer class
- removed COWOA pages for downloads, since tidied template no longer needs them

### CODE REVIEW

File: includes/modules/pages/download/header_php.php [hunk 1]
```diff
@@ -28,245 +29,183 @@
...
+  header("Last-Modified: " . gmdate("D,d M Y H:i:s") . " GMT");
+  switch($detectedBrowser)
+  {
+    case 'IE5':
+    case 'IE6':
+      header("Pragma: public");
+      header("Cache-Control: must-revalidate, post-check=0, pre-check=0");
+      header("Cache-Control: private", FALSE);
+      header("Cache-Control: max-age=1");  // stores for only 1 second, which helps allow SSL downloads to work more reliably in IE
+    break;
```
Comment (lhungil) [inline includes/modules/pages/download/header_php.php]: As lines 175 - 178 are the same as 183-186, would suggest further refactoring to remove the duplicate code.

**Comment by drbyte** [comment]
Notes: 
- If using with Dropbox, set the `&dl=0` to `&dl=1` on the "sharing link" that Dropbox gives you.
- For AWS, to keep credentials out of version-control, put them into the extra-configures folder, and prefix the filename with `dev-` since that's in the project's .gitignore already.

**Comment by zcwilt** [comment]
:+1:

**Comment by ajeh** [comment]
:+1:

**Pull request merged**